### DISCUSSION TITLE
Installation instructions for Linux Mint Vera

### DISCUSSION THREAD

**Discussion by Tokn59** [abraunegg/onedrive#2372] [General]
Hello,

I am using the following distri:

Distributor ID:	Linuxmint
Description:	Linux Mint 21.1
Release:	21.1
Codename:	vera

This version is not mentioned explicitly in the list of versions for instructions. The closest I see is Linux Mint 21.x / Vanessa

So is it OK to use that version and thus use the instructions for Distribution: Ubuntu 22.04 ?

Thank you !!!!

**Comment by abraunegg** [comment]
@Tokn59 
> So is it OK to use that version and thus use the instructions for Distribution: Ubuntu 22.04 ?

Linux Mint 21.1 'Vera' is based on Ubuntu 22.04 - so the instructions for Ubuntu 22.04 are appropriate to use.

**Reply by gitkevin** [reply]
I'm also running Linux Mint 21.1, and used Ubuntu 22.04 instructions. Step 4 failed for me, as below.

```
sudo apt install --no-install-recommends --no-install-suggests onedrive 

The above ‘apt install’ resulted in the error messages below.

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:
The following packages have unmet dependencies:
onedrive : Depends: libphobos2-ldc-shared94 (>= 1:1.24.0) but it is not installable
E: Unable to correct problems, you have held broken packages.
```

At this point I'm wondering whether it would be easier to grab a Docker image which uses a Linux version that supports the needs of onedrive better. Looking at the table on the [INSTALL.md page](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md) it looks like Fedora could be a good choice. From the [Docker page](https://hub.docker.com/_/fedora), it looks like Fedora 38 is mature enough. [Fedora looks to support onedrive well](https://koji.fedoraproject.org/koji/packageinfo?packageID=26044).

Does anyone have comments (especially comments based on experience with something like what I'm proposing to myself) on the merits of this approach?

Thank you....

P.S.:  I hope I didn't hijack this thread. I'm hoping that other Linux Mint 21.1 users might benefit from the proximity of the similar comments.

**Comment by abraunegg** [comment]
@gitkevin 

What version of `libphobos2-ldc-shared` is actually installable on your version of Linux Mint?

libphobos2-ldc-shared94 was from Debian Bullseye (oldstable) - https://packages.debian.org/bullseye/libphobos2-ldc-shared94

When these package installation problems pop-up - 99.99% of the time folk are following the incorrect installation instructions. They think that they are running version X of their OS .. when they are not - thus, they follow the wrong set of instructions and get this issue.

Are you 100% sure you are running Linux Mint 21.1 and not potentially [Linux Mint Debian Edition 5](https://linuxmint.com/rel_elsie.php) which was based on Debian Bullseye? If this is the case, you *must* follow the [Debian 11 instructions](https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-debian-11) to install the client.

In terms of Linux Mint versions - this is what the [official documentation](https://linuxmint.com/download_all.php) maps to:

![image](https://github.com/abraunegg/onedrive/assets/4956234/75e69a05-3b44-4e06-bd07-32ca06c33a6b)

When I use Linux Mint 21.1 instructions on Linux Mint 21.1 there is no issue:

**Installation:**
![image](https://github.com/abraunegg/onedrive/assets/4956234/93aa55d8-040e-4bb7-85fc-536043e6d263)

![image](https://github.com/abraunegg/onedrive/assets/4956234/2baf8282-bb73-42aa-9ae5-ea14f934ec25)

As per above, Linux Mint 21.1 requires `libphobos2-ldc-shared98` .....

![image](https://github.com/abraunegg/onedrive/assets/4956234/5c0f0ac3-8e49-40ce-b660-ec7d664c813f)

So please can you:
* Double check what Linux Mint version you are running
* Read the installation instructions more carefully

**Comment by gitkevin** [comment]
**I determined my Linux version like so. This is also "Step 3: Determine what your OS is based on"**

```
gitkevin:~/temp$ lsb_release --all
No LSB modules are available.
Distributor ID:	Linuxmint
Description:	Linux Mint 21.1
Release:	21.1
Codename:	vera
```
I don't know if it makes a difference, but I'm running the Cinnamon variant of Linux Mint 21.1.

**Below is what I wrote down as I was going through the process.**


**Step 1: Step 1: Remove any configured PPA and associated 'onedrive' package and systemd service files**

```
sudo apt remove onedrive
sudo add-apt-repository --remove ppa:yann1ck/onedrive

sudo rm /etc/systemd/user/default.target.wants/onedrive.service
```

**Step 2: Ensure your system is up-to-date***

```
sudo rm -rf /var/lib/dpkg/lock-frontend
sudo rm -rf /var/lib/dpkg/lock
sudo apt-get update
sudo apt-get upgrade
sudo apt-get dist-upgrade
sudo apt-get autoremove
sudo apt-get autoclean
```

I answered yes to rm's questions, rather than use -y without knowing what it was going to do.

```reboot```

**Step 3: Determine what your OS is based on**
See the top of this comment. That's what I did for this step.

**[Step 4: Pick the correct instructions to use]**(https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#step-4-pick-the-correct-instructions-to-use)

Runing Linux Mint 21.x implies to use [Ubuntu 22.04](https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2204) instructions.

**Step 1: Add the OpenSuSE Build Service repository release key** 

wget -qO - https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_22.04/Release.key | gpg --dearmor | sudo tee /usr/share/keyrings/obs-onedrive.gpg > /dev/null

**Step 2: Add the OpenSuSE Build Service repository** 

echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/obs-onedrive.gpg] https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_21.10/ ./" | sudo tee /etc/apt/sources.list.d/onedrive.list

**Step 3: Update your apt package cache** 

sudo apt-get update

Hours earlier today I attended to a kernel update, and I ran apt-get update and apt-get upgrade and apt-get autoremove, all without incident. The subsequent update was fine as well.

**Step 4: Install 'onedrive'** 

sudo apt install --no-install-recommends --no-install-suggests onedrive 

The above ‘apt install’ resulted in the error messages below.

```
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:
The following packages have unmet dependencies:
onedrive : Depends: libphobos2-ldc-shared94 (>= 1:1.24.0) but it is not installable
E: Unable to correct problems, you have held broken packages.
```

**Step 5: Read** '[Known Issues](https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#known-issues-with-installing-from-the-above-packages)' **with these packages**
The known issues had to do with specific situations where there could be a core dump from onedrive. That wasn't my issue.


You might have noticed that I pretty much copied & pasted my commands from [INSTALL.md](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md) and [ubuntu-package-install.md](https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#distribution-ubuntu-2204).  Those "copy" widgets are very helpful!

Did you spot anything that would trigger the error messages I reported?

Thanks...

**Comment by abraunegg** [comment]
@gitkevin 

Yes .. you are adding the *wrong* repository as part of step 2 .. this is why it is failing

**Reply by gitkevin** [reply]
I see it, 21.10.  Ugh!  Thanks!
